File: app/src/main/java/com/github/anastr/speedview/AwesomeSpeedometerActivity.java
package com.github.anastr.speedview;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.SeekBar;
import android.widget.TextView;

import com.github.anastr.speedviewlib.AwesomeSpeedometer;

import java.util.Locale;

public class AwesomeSpeedometerActivity extends AppCompatActivity {

    AwesomeSpeedometer awesomeSpeedometer;
    SeekBar seekBar;
    Button speedTo, realSpeedTo, stop;
    TextView textSpeed;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_awesome_speedometer);

        awesomeSpeedometer = findViewById(R.id.awesomeSpeedometer);
        seekBar = findViewById(R.id.seekBar);
        speedTo = findViewById(R.id.speedTo);
        realSpeedTo = findViewById(R.id.realSpeedTo);
        stop = findViewById(R.id.stop);
        textSpeed = findViewById(R.id.textSpeed);

        speedTo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                awesomeSpeedometer.speedTo(seekBar.getProgress());
            }
        });

        realSpeedTo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                awesomeSpeedometer.realSpeedTo(seekBar.getProgress());
            }
        });

        stop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                awesomeSpeedometer.stop();
            }
        });

        seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                textSpeed.setText(String.format(Locale.getDefault(), "%d", progress));
            }

            @Override
            public void onStartTrackingTouch(SeekBar seekBar) {
            }

            @Override
            public void onStopTrackingTouch(SeekBar seekBar) {
            }
        });
    }
}

File: app/src/main/java/com/github/anastr/speedviewapp/AwesomeSpeedometerActivity.kt
package com.github.anastr.speedviewapp

import android.os.Bundle
import android.widget.Button
import android.widget.SeekBar
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.github.anastr.speedviewlib.AwesomeSpeedometer
import java.util.*

class AwesomeSpeedometerActivity : AppCompatActivity() {

    lateinit var textSpeed: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_awesome_speedometer)
        title = "Awesome Speedometer View"

        val awesomeSpeedometer = findViewById<AwesomeSpeedometer>(R.id.awesomeSpeedometer)
        val seekBar = findViewById<SeekBar>(R.id.seekBar)
        val speedTo = findViewById<Button>(R.id.speedTo)
        val realSpeedTo = findViewById<Button>(R.id.realSpeedTo)
        val stop = findViewById<Button>(R.id.stop)
        textSpeed = findViewById(R.id.textSpeed)
        awesomeSpeedometer.withTremble = false

        speedTo.setOnClickListener { awesomeSpeedometer.speedTo(seekBar.progress.toFloat()) }

        realSpeedTo.setOnClickListener { awesomeSpeedometer.realSpeedTo(seekBar.progress.toFloat()) }

        stop.setOnClickListener { awesomeSpeedometer.stop() }

        seekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                textSpeed.text = "%d".format(Locale.getDefault(), progress)
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}

            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
    }
}
